src/frequenz/client/electricity_trading/_client.py

```diff
@@ -47,6 +48,35 @@ class _Sentinel:
 
 
 NO_VALUE = _Sentinel()
+PRECISION_DECIMAL_PRICE = 2
+PRECISION_DECIMAL_QUANTITY = 1
+
+
+def validate_decimal_places(value: Decimal, decimal_places: int, name: str) -> None:
+    """
+    Validate that the decimal places of a given value do not exceed a specified limit.
+
+    Args:
+        value: The value to be checked.
+        decimal_places: The maximum allowed decimal places.
+        name: The name of the value (for error messages).
+
+    Raises:
+        ValueError: If the value has more decimal places than allowed.
+        ValueError: If the value is not a valid decimal number.
+    """
+    assert decimal_places >= 0, "The decimal places must be a non-negative integer."
+
+    try:
+        exponent = int(value.as_tuple().exponent)
+        if abs(exponent) > decimal_places:
+            raise ValueError(
+                f"The {name} cannot have more than {decimal_places} decimal places."
+            )
+    except InvalidOperation as exc:
+        raise ValueError(
+            f"{value} cannot have more than {decimal_places} decimal places"
```

Comment:
I think the error message here is not entirely right, maybe something like `f"The {value} for {name} is not a valid decimal number.`